src/Microsoft.ML.Core/EntryPoints/ComponentFactory.cs

```diff
@@ -64,4 +64,32 @@ public interface IComponentFactory<in TArg1, in TArg2, out TComponent> : ICompon
     {
         TComponent CreateComponent(IHostEnvironment env, TArg1 argument1, TArg2 argument2);
     }
+
+    /// <summary>
+    /// An interface for creating a component when we take three extra parameters (and an <see cref="IHostEnvironment"/>).
+    /// </summary>
+    public interface IComponentFactory<in TArg1, in TArg2, in TArg3, out TComponent> : IComponentFactory
+    {
+        TComponent CreateComponent(IHostEnvironment env, TArg1 argument1, TArg2 argument2, TArg3 argument3);
+    }
+
+    /// <summary>
+    /// A class for creating a component when we take three extra parameters
+    /// (and an <see cref="IHostEnvironment"/>) that simply wraps a delegate which
+    /// creates the component.
+    /// </summary>
+    public class SimpleComponentFactory<TArg1, TArg2, TArg3, TComponent> : IComponentFactory<TArg1, TArg2, TArg3, TComponent>
```

Comment:
>SimpleComponentFactory [](start = 17, length = 22)

Thanks @eerhardt. As we discussed in #681 perhaps we can hide these implementations of the interface, but we'll do that after we merge this I suppose.

Reply:
Yes, I will take care of all the `SimpleComponentFactory` classes being hidden in #681.